platforms/documentation/docs/src/docs/userguide/dep-man/basics/dependency_configurations.adoc

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2023 Gradle, Inc.
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[dependency-configurations]]
+= 2. Dependency Configurations
+
+Every dependency declared for a Gradle project applies to a specific scope.
+
+For example, some dependencies should be used for compiling source code whereas others only need to be available at runtime:
+
+[source, kotlin]
+.build.gradle.kts
+----
+dependencies {
+    implementation("com.google.guava:guava:30.0-jre") // Needed to compile and run the app
+    runtimeOnly("org.json:json:20220924") // Only needed at runtime
+}
+----
+
+Dependency configurations are a way to define different sets of dependencies for different purposes within a project.
+They determine how and when dependencies are used in various stages of the build process.
+
+Configurations are a fundamental part of dependency resolution in Gradle.
+
+[[sec:what-are-dependency-configurations]]
+== Understanding dependency configurations
+
+Gradle represents the scope of a dependency with the help of a link:{groovyDslPath}/org.gradle.api.artifacts.Configuration.html[Configuration].
+Every configuration can be identified by a unique name.
+
+Many Gradle plugins add pre-defined configurations to your project.
+
+The <<java_plugin.adoc#sec:java_plugin_and_dependency_management,Java plugin>>, which is used for building Java projects and provides a way to compile, package, and test Java applications, adds many configurations.
```

Comment:
I'm not sure how I feel about listing the Java-specific stuff here. I would assume this content belongs in the java section. 

However, if we do list this content here, we should link to the `Java Library` plugin, not just the java plugin, which also defines `api` and `compileOnlyApi` configurations as well as the ones listed here.

Reply:
It's tough to describe configurations without an example. What do you think of adding a c++ or swift example?

Reply:
I updated the doc to use the `Java Library` plugin. I think it's ok to focus on Java since its still the most common Gradle use and its the easiest way to introduce this concept.